### PR TITLE
Linkify comments

### PR DESCRIPTION
**Pull request by aakash2330**
Hyperlink Links in Comments #431

Turn Links within comments clickable . 
(Please ignore the first commit )

### CODE REVIEW

File: apps/web/ui/links/linkify-comments.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+//to convert links in comments clickable
+export default function LinkifyText({ children }:{children:string}) {
+  const urlPattern = /(\bhttps?:\/\/[-A-Z0-9+&@#/%?=~_|!:,.;]*[-A-Z0-9+&@#/%=~_|])/gi;
+
+  const renderText = (text:string) => {
+    const parts = text.split(urlPattern);
+    return parts.map((part, index) => {
+      if (part.match(urlPattern)) {
+        return <a href={part} key={index} target="_blank" rel="noopener noreferrer" className="hover:underline">{part}</a>;
```
Comment (tuffz) [inline apps/web/ui/links/linkify-comments.tsx]: I would also recommend adding a "nofollow" to the `rel` attribute to avoid link spammers.

**Review by unrenamed — CHANGES_REQUESTED** [review]
Hey @aakash2330, excited to help with your PR code review!

1) Thinking about making the `<LinkifyText />` component even more modular by moving it to the `packages/ui` module. What do you think? Also, we can tweak the filename to match the component name, maybe something like `linkify-text`.

2) Noticed the #431 issue author mentioned an ideal scenario: allowing URLs without protocols to transform into hyperlinks. 

   > ideally not even requiring entering https:// but open to having that if it makes this easier

    Currently, it seems to support only `http(s)` protocols. Any challenges with this? Maybe check out existing solutions like [linkify.js.org](https://linkify.js.org/)? Let me know your thoughts! 😊

**Comment by aakash2330** [comment]
Hey @tuffz @unrenamed thanks for reviewing my pr . 

1 . Added the "nofollow" to the rel attribute as suggested by @tuffz .
2 . Tweaked the filename to match the component name (linkify-comments) and moved it to packages/ui module .
3. Want to get up a heads up from the maintainers before including a third party library that's why I didn't include it in the first place .

**Comment by steven-tey** [comment]
Thanks a lot for the PR @aakash2330 + feedback from @tuffz & @unrenamed 🙏 

re: `linkifyjs` – it looks lightweight enough, let's use it 😄 (and rely on @socketdev for 3rd party deps security 🔒)

**Comment by aakash2330** [comment]
@steven-tey added linkifyjs to support more URL protocols as well as allowing URLs without protocols to transform into hyperlinks.

**Comment by steven-tey** [comment]
Great work on this PR @aakash2330 – also shoutout to @unrenamed for the linkify tip as well!

Will be merging this now – also updated our [changelog](https://dub.co/changelog/link-comments) & [help docs](https://dub.co/help/article/how-to-create-link#link-comments) to include this too :)

**Pull request merged**